### DISCUSSION TITLE
Logging config from within task function

### DISCUSSION THREAD

**Discussion by dkajtoch** [mit-ll-responsible-ai/hydra-zen#629] [Q&A] [answered]
When using hydra I was able to log (mlflow, w&b) all my parameters from the DictConfig before training started (since they were passed to the function). With hydra-zen I do not see a way to access the config that was used to instantiate objects in the task function. Is there a way to get DictConfig before executing the training (for logging purposes with all overrides)?

```
store.add_to_hydra_store()
task_function.hydra_main(config_path=None, config_name="config", version_base="1.2")
```

**Comment by dkajtoch** [comment]
I see now that there is extra variable `zen_cfg` passed to the task function which is exactly what I am looking for!

**Reply by rsokl** [reply, marked as answer]
You might also consider using `zen`'s `pre_call` argument. This accepts a function that is passed the full config before calling the main task function that you are using:

```python
def log_config(cfg): ...
def main_fn(x, y, z): ...

task_function = zen(main_fn, pre_call=log_config)
```

The advantage to this is that you don't need to modify `main_fn`'s signature/body to add logging and you can use `log_config` across applications.

**Comment by rsokl** [comment]
(reopening to improve visibility for others)
